posthog/temporal/batch_exports/temporary_file.py

```diff
@@ -458,10 +458,15 @@ def write_dict(self, d: dict[str, typing.Any]) -> int:
         try:
             n = self.batch_export_file.write(orjson.dumps(d, default=str) + b"\n")
         except orjson.JSONEncodeError:
+            logger.exception("Failed to encode with orjson: %s", d)
             # orjson is very strict about invalid unicode. This slow path protects us against
             # things we've observed in practice, like single surrogate codes, e.g. "\ud83d"
             cleaned_content = replace_broken_unicode(d)
             n = self.batch_export_file.write(orjson.dumps(cleaned_content, default=str) + b"\n")
+        except TypeError:
+            logger.exception("Orjson detected a deeply nested dict: %s", d)
+            raise
```

Comment:
Sometimes users request AI to return JSON and they put the entire thing as a property. If it's some very nested (>255) garbage, then orjson raises this error.